WearOAuth/util/src/main/java/com/example/android/wearable/oauth/util/Requests.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.wearable.oauth.util
+
+import android.util.Log
+import org.json.JSONObject
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URL
+import java.net.URLEncoder
+
+/**
+ * Simple implementation of a POST request. Normally you'd use a library to do these requests.
+ */
+fun doPostRequest(url: String, params: Map<String, String>): Result<JSONObject> {
```

Comment:
Since this function has a return type of `Result`, I'd expect that this method wouldn't throw things like `IOException`.

Reply:
I've worked on this a bit more. I tried making these methods return a Result & making them suspending. However, it kept giving weird runtime errors ("can't cast Result.Failure to Result") so I ended up moving this back to an exception-throwing version.